docs/contributing.md

```diff
@@ -37,85 +37,54 @@ your GitHub username:
 $ git clone https://github.com/YOUR-USERNAME/httpx
 ```
 
-With the repository cloned you can access its folder, set up the
-virtual environment, install the project requirements,
-and then install HTTPX on edit mode:
+You can now install the project and its dependencies using:
 
 ```shell
 $ cd httpx
-$ python3 -m venv venv
-$ source venv/bin/activate
-$ pip install -r test-requirements.txt
-$ pip install -e .
+$ scripts/install
 ```
 
-!!! note
-    Feel free to replace this step with your development environment setup
-    (pyenv, pipenv, virtualenvwrapper, docker, etc).
-
 ## Testing and Linting
 
-We use [nox](https://nox.thea.codes/en/stable/) to automate testing, linting,
-and documentation building workflow. Make sure you have it installed
-at your system before starting.
-
-Install `nox` with:
-
-```shell
-$ python3 -m pip install --user nox
-```
-
-Alternatively, use [pipx](https://github.com/pipxproject/pipx) if you prefer
-to keep it into an isolated environment:
-
-```shell
-$ pipx install nox
-```
+We use custom shell scripts to automate testing, linting,
+and documentation building workflow.
 
-Now, with nox installed, run the complete pipeline with:
+To run the tests, use:
 
 ```shell
-$ nox
+$ scripts/test
 ```
 
 !!! warning
-    The test suite spawns a testing server at the port **8000**.
-    Make sure this isn't being used, so the tests can run properly.
+    The test suite spawns testing servers on ports **8000** and **8001**.
+    Make sure these are not in use, so the tests can run properly.
 
-To run the code auto-formatting separately:
+You can run a single test script like this:
 
 ```shell
-$ nox -s lint
+$ scripts/test -- tests/test_multipart.py
 ```
 
-Also, if you need to run the tests only:
+To run the code auto-formatting:
 
 ```shell
-$ nox -s test
+$ scripts/lint
 ```
 
-You can also run a single test script like this:
+Lastly, to run code checks separately (they are also run as part of `scripts/test`), run:
 
 ```shell
-$ nox -s test -- tests/test_multipart.py
+$ scripts/check
 ```
 
-Lastly, to ensure you're on track to pass the CI build, run:
-
-```shell
-$ scripts/test
-```
-
-This command is a light wrapper around `nox` that will run code style checks and test the code against all installed Python versions.
-
 ## Documenting
 
 Documentation pages are located under the `docs/` folder.
 
 To run the documentation site locally (useful for previewing changes), use:
 
 ```shell
-$ nox -s serve
+$ scripts/serve
```

Comment:
I’m guessing we want `scripts/docs serve` here?

Reply:
What would we think about `scripts/docs-serve` and `scripts/docs-build`? (We'd even get autocomplete without having to deal with CLI arguments in plain bash.)

Reply:
Also the `serve` script is actually missing right now. 😆